src/main/java/com/synopsys/integration/jenkins/scan/ScanPipelineCommands.java

```diff
@@ -46,23 +57,61 @@ public int runScanner(Map<String, Object> scanParameters, ScanStrategyFactory sc
                 bridgeDownloadManager.initiateBridgeDownloadAndUnzip(bridgeDownloadParams);
             } else {
-                listener.getLogger().println("Bridge download is not required. Found installed in: " + bridgeDownloadParams.getBridgeInstallationPath());
+                logger.info("Bridge download is not required. Found installed in: " + bridgeDownloadParams.getBridgeInstallationPath());
+                logger.println(LogMessages.DASHES);
             }
             FilePath bridgeInstallationPath = new FilePath(workspace.getChannel(), bridgeDownloadParams.getBridgeInstallationPath());
 
             try {
                 exitCode = scanner.runScanner(scanParameters, scanStrategy, bridgeInstallationPath);
             } catch (Exception e) {
-                e.printStackTrace(listener.getLogger());
                 throw new ScannerJenkinsException(ExceptionMessages.scannerFailureMessage(e.getMessage()));
             }
         }
 
-        if (exitCode != 0) {
-            throw new ScannerJenkinsException(ExceptionMessages.scannerFailedWithExitCode(exitCode));
+        try {
+            if (exitCode != 0) {
+                throw new ScannerJenkinsException(ExceptionMessages.scannerFailedWithExitCode(exitCode));
+            }
+        }
+        finally {
+            logger.println("**************************** END EXECUTION OF SYNOPSYS SECURITY SCAN ****************************");
         }
 
         return exitCode;
     }
 
+    public void logMessagesForParameters(Map<String, Object> scanParameters) {
+        logger.println("-------------------------- Parameter Validation Initiated --------------------------");
+
+        String scanType = ((ScanType) scanParameters.get(ApplicationConstants.SCAN_TYPE_KEY)).toString().toLowerCase();
+
+        logger.info(" --- " + ApplicationConstants.SCAN_TYPE_KEY + " = " + scanType);
+        logger.info("parameters for %s:", scanType);
+
+        for (Map.Entry<String, Object> entry : scanParameters.entrySet()) {
+            String key = entry.getKey();
+            if(key.contains(scanType)) {
+                if(key.equals(ApplicationConstants.BRIDGE_BLACKDUCK_API_TOKEN_KEY) || key.equals(ApplicationConstants.BRIDGE_POLARIS_ACCESS_TOKEN_KEY) || key.equals(ApplicationConstants.BRIDGE_COVERITY_CONNECT_USER_PASSWORD_KEY)) {
+                    entry.setValue(LogMessages.ASTERISKS);
+                }
+                Object value = entry.getValue();
+                logger.info(" --- " + key + " = " + value.toString());
+            }
+
+        }
+
+        logger.println(LogMessages.DASHES);
+
+        logger.info("parameters for bridge:");
+
+        for (Map.Entry<String, Object> entry : scanParameters.entrySet()) {
+            String key = entry.getKey();
+            if(key.equals(ApplicationConstants.BRIDGE_DOWNLOAD_URL) || key.equals(ApplicationConstants.BRIDGE_DOWNLOAD_VERSION) || key.equals(ApplicationConstants.BRIDGE_INSTALLATION_PATH)) {
```

Comment:
`key.equals(ApplicationConstants.INCLUDE_DIAGNOSTICS_KEY)` missing

Reply:
Yeah. Added it.